app/controllers/api/functions.php

```diff
@@ -556,7 +556,8 @@
             // Remove deploy for all other deployments.
             $deployments = $dbForProject->find('deployments', [
                 new Query('deploy', Query::TYPE_EQUAL, [true]),
-                new Query('functionId', Query::TYPE_EQUAL, [$functionId])
+                new Query('resourceId', Query::TYPE_EQUAL, [$functionId]),
+                new Query('resourceType', Query::TYPE_EQUAL, ['functions'])
```

Comment:
Since we're also querying by `resource` type, I think we need an index for this field as well.